test/integration_test.py

```diff
@@ -2093,3 +2093,18 @@ def test_response_security_headers(self):
                         response.raise_for_status()
                         expected = expected_headers | global_headers
                         self.assertIsSubset(expected.items(), response.headers.items())
+
+    def test_default_4xx_response_headers(self):
+        headers = {
```

Comment:
This is (or should be) identical to the above dictionary literal. Please extract the above variable to a class attribute and import and use that here.

Furthermore, there are a bunch of related literals in similar integration tests (look for global_headers) and in chalice.py. I think these can and should be backed by that same attribute. Please perform this refactoring in a separate commit.